src/tools/perf/lib/libperf_thread.c

```diff
@@ -30,17 +30,19 @@ static ucs_status_t ucx_perf_thread_run_test(void* arg)
     ucx_perf_params_t* params       = &perf->params;
     ucs_status_t status;
 
-    /* new threads need explicit device association */
-    status = perf->send_allocator->init(perf);
-    if (status != UCS_OK) {
-        goto out;
-    }
-
-    if (perf->send_allocator != perf->recv_allocator) {
-        status = perf->recv_allocator->init(perf);
+    if (params->api == UCX_PERF_API_UCT) {
+        /* new threads need explicit device association */
+        status = perf->send_allocator->init(perf);
         if (status != UCS_OK) {
             goto out;
         }
+
+        if (perf->send_allocator != perf->recv_allocator) {
+            status = perf->recv_allocator->init(perf);
+            if (status != UCS_OK) {
+                goto out;
+            }
+        }
```

Comment:
can we extract common code with ucx_perf_allocators_init()?
the common function could be called ucx_perf_allocators_init_thread()